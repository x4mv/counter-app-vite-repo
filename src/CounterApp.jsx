import {useState} from 'react'
import PropTypes from 'prop-Types'



export const CounterApp = ({value}) => {
    
    console.log('render');



    const [counter, setCounter] = useState(value) 



    // Uso de useState
    const addCounter = ( ) => setCounter(counter + 1);
    const diminish = () => setCounter(counter - 1 );
    const reset = () => setCounter(value);


    return (
        <>
            <h1>CounterApp</h1>
            <h2> { counter } </h2>

            <button onClick={addCounter}>+1</button>            
            <button onClick={diminish}>-1</button>
            <button aria-label="btn-reset" onClick={reset}>Reset</button>

        </>
    )
}

CounterApp.propTypes = {
    value: PropTypes.number.isRequired
}
