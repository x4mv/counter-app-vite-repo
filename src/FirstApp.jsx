import PropTypes from 'prop-Types'

export const FirstApp = ( {title, subtitle} ) => {

  
if (!title) {
  throw new Error('El titulo no existe');
}

  return (
    <>
    <h1>{ title }</h1>
    <p>{ subtitle }</p>
    </>
  )
}


FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.any
}

FirstApp.defaultProps = { 
  title: 'No hay titulo', 
  subtitle: 'El es un estudiante de la UNA'
}