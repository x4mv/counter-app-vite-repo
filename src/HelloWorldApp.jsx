import React from 'react'

/* function App(){
    return (<h1>Hola mundo</h1>)
}

export default App; */


export const HelloWorldApp = () => {
  return (
    <div><h1>Hello World</h1></div>
  );
}
