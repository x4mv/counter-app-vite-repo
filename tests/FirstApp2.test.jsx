import {  render, screen } from "@testing-library/react"
import { FirstApp } from "../src/FirstApp"

describe('pruebas en FirstApp', () => { 

    const title = 'Hola soy Sergio';
    const subtitle = 'Es un subtitulo'

    test('Hacaer match con el snapshot', () => { 
        const { container } = render(<FirstApp title = {title}/>);
        expect( container ).toMatchSnapshot();
    });


    test('debe de mostrar el mensaje "Hola, soy Sergio"', () =>{
        render( <FirstApp title={ title }/>);

        expect(screen.getByText(title)).toBeTruthy();
        // screen.debug(); para ver el obj
        
    });

    test('Debe de mostrar el titulo en un h1', () => { 
        render( <FirstApp title={title}/>)
        expect( screen.getByRole('heading', {level: 1 }).innerHTML).toContain(title);
    });

    test('Debe de mostrar el subtitulo enviado por props', () => { 
        
        render( <FirstApp 
            title={ title }
            subtitle={subtitle}
        />);

        expect( screen.getAllByText(subtitle).length).toBe(3)


    });
});