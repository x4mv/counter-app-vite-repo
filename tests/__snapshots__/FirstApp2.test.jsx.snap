// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en FirstApp Hacaer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hola soy Sergio
  </h1>
  <p>
    El es un estudiante de la UNA
  </p>
  <p>
    El es un estudiante de la UNA
  </p>
  <p>
    El es un estudiante de la UNA
  </p>
</div>
`;
